packages/react-router/src/types.ts

```diff
@@ -57,3 +57,5 @@ export interface RouteDefinition {
   render?(details: RouteRenderDetails): ReactNode;
   renderPrefetch?(details: RouteRenderPrefetchDetails): ReactNode;
 }
+
+export type RelativeTo = 'root' | string;
```

Comment:
I think we should just make this `'root' | 'prefix'`. `root` gives the new behavior, `prefix` gives the existing behavior. I think I will eventually add a third option, `'current'`, to make the navigation relative to the full current path, but for now I think just supporting the two options, with room for the third, is great.